components/sysutils/nut/files/nut.in

```diff
@@ -0,0 +1,84 @@
+#!/sbin/sh
+
+# Trivial (better is yet to come) SMF method script to start nut services
+# Adapted for OpenIndiana userland from init.d script template in NUT sources
+# Adaptation copyright (C) 2016 Jim Klimov
+
+if [ -z "$SMF_FMRI" ]; then
+    echo "$0 must be called in SMF context!" &>2
+    exit 1
+fi
+
+# smf(5)
+. /lib/svc/share/smf_include.sh || exit
+
+NUT_DIR="@USRDIR@"
+NUT_SBIN_DIR="$NUT_DIR/sbin"
+#NUT_SBIN_DIR="$NUT_DIR/sbin/@MACH64@"
+NUT_RUN_DIR="/@NUTRUNDIR@"
+NUTUSER="@NUTUSER@"
+NUTGROUP="@NUTGROUP@"
+CONFIG="/@NUTETCDIR@/nut.conf"
+
+if [ -f "$CONFIG" ] ; then
+	. "$CONFIG"
+fi
+
+ups_stop () {
+	pkill -n upsmon
+	pkill -n upsd
+	"${NUT_SBIN_DIR}"/upsdrvctl stop > /dev/null 2>&1
+}
+
+ups_start () {
+	if [ "$MODE" = "none" ];then
+		echo "No NUT mode set" >&2
+		exit $SMF_EXIT_ERR_CONFIG
+	fi
+
+	# Default rights inspired by NUT scripts/Solaris/postinstall.in
+	mkdir -p "$NUT_RUN_DIR" && \
+	chown "root:$NUTGROUP" "$NUT_RUN_DIR" && \
+	chmod 770 "$NUT_RUN_DIR" \
+	|| exit $SMF_EXIT_ERR_FATAL
+
+	if [ ! "$MODE" = "netclient" ];then
+		"${NUT_SBIN_DIR}"/upsdrvctl start #> /dev/null 2>&1
+		"${NUT_SBIN_DIR}"/upsd #> /dev/null 2>&1
+	fi
+	"${NUT_SBIN_DIR}"/upsmon #> /dev/null 2>&1
+}
+
+case "$1" in
+'start')
+	ups_start
+	;;
+
+'stop')
+	ups_stop
+	;;
+
+'restart')
+	ups_stop
+	while pgrep upsd > /dev/null
+	do
+		sleep 1
+	done
+	ups_start
+	;;
+'poweroff')
```

Comment:
Probably not, but we need the script delivered somewhere for the usecase of shutdowns that tell the UPS to power itself off (and the UPSes that can do so). Since in this version the SMF method script is a wraparound of the initscript, this stayed here.

I think later we can e.g. make an SMF instance (`nut:poweroff`?) whose `exec start` invocation as part of dependencies for the system to go *down* (optional-required by milestone that shuts down) would be to run `/lib/svc/method/svc-nut poweroff`, and the service and/or script should also require presence of `/etc/nut/killpower` flagfile set by `upsmon`.

Also users can still call it directly, just pre-exporting an SMF_FMRI value to get around the checks on top.